add tls  examples (#108)
streamnative/examples: tls/go/README.md

# Overview

> This document describes how to produce message to and consume message from a Pulsar cluster using the [Go client](https://github.com/apache/pulsar-client-go).

Reviewer: ```suggestion
This document describes how to produce messages to and consume messages from a Pulsar cluster using the [Go client](https://github.com/apache/pulsar-client-go).
```

# Prerequisites

- Go 1.11 or higher version
- Go client 0.1.1+ (without 0.1.1)

If you have not installed Go, install it according to the [installation instruction](http://golang.org/doc/install).

# Example

**Note**

> This example shows how to connect to a Pulsar cluster through the OAuth2 authentication plugin. To connect to a Pulsar cluster through the Token authentication plugin, see the implementation of [connectByToken.go](https://github.com/streamnative/pulsar-examples/blob/master/cloud/go/connectByToken.go).

In this example, the Go producer publishes data to the `topic-1` in your Pulsar cluster. The consumer receives the message from the `topic-1` and acknowledges each received message.

> The content of each message payload is a combination of `hello-` and a digital (0-9), such as `hello-0`).

Reviewer: ```suggestion
The content of each message payload is a combination of `hello-` and a digit(0-9), e.g., `hello-0`.
```
